FullStorySegmentMiddleware/Classes/FullStoryMiddleware.m

```diff
@@ -126,88 +130,80 @@ - (SEGPayload *) getNewPayloadWithFSURL:(SEGContext * _Nonnull)context {
+                        // get the current array form this suffix, if any, then append current item
+                        NSMutableArray *tempArr = [[NSMutableArray alloc] initWithArray:[props valueForKey:[key stringByAppendingString:suffix]]];
+                        [tempArr addObject:item];
+                        [props setValue:tempArr forKey:[key stringByAppendingString:suffix]];
                     }
                 }
-                [props setValue:arr forKey:[key stringByAppendingString:suffix]];
                 [props setValue:dict forKey:[key stringByReplacingOccurrencesOfString:@"_" withString:@"-"]];
             }else{
                 suffix = [self getSuffixStringFromObject:obj];
+                
                 [props setValue:[properties valueForKey:key] forKey:[key stringByAppendingString:suffix]];
             }
         }else{
-            NSLog(@"is not object");
+            #ifdef DEBUG
+                NSAssert(FALSE, @"key `%@` is not an object can't be serialized for FS custom event.", key);
+            #else
+                NSLog(@"key `%@` is not an object can't be serialized for FS custom event.", key);
+                // if prod, then don't send it
+            #endif
         }
-        
-        NSLog(@"new props is %@", props);
-        
-        
-        //
-        //        // handle premitive types:
-        //               if ( NSNotFound != [[NSString stringWithFormat:@"%s", typeCode] rangeOfCharacterFromSet:[NSCharacterSet characterSetWithCharactersInString:@"{}"]].location ) {
-        //                   //this value is struct, do not handle
-        //               }
-        //               else if ( [@"i" isEqualToString:@(typeCode)] )
-        //               {
-        //                   //this value is int
-        //               }
-        //
-        //        // do type check if there is underscore
-        //               if([key containsString:@"_"]){
-        //                   // FS requirements: https://help.fullstory.com/hc/en-us/articles/360020623234-FS-Recording-Client-API-Requirements
-        //                   NSLog(@"%@ contains _",key);
-        //                   if([[props valueForKey:key] isKindOfClass:[NSString class]]){
-        //
-        //                   }else if([[props valueForKey:key] isKindOfClass:[NSNumber class]]){
-        //                       NSLog(@"%@ is number",key);
-        //                   }
-        //
-        //               }
     }
-    
-    
-    
-    
     return props;
 }
 
 - (NSString *) getSuffixStringFromObject: (NSObject *) obj{
     // defualt to string
     NSString *suffix = @"_str";
-    
     if([obj isKindOfClass:[NSNumber class]]){
+        // defaut to real
         suffix = @"_real";
-    }else if([obj isKindOfClass:[NSString class]]){
-        suffix = @"_str";
+        NSNumber *n = (NSNumber *) obj;
+        const char *type = n.objCType;
```

Comment:
Other places in the code `@encode` is used. This should also use `@encode` for consistency.